server/src/controllers/teamController.js

```diff
@@ -15,33 +15,35 @@ const addTeam = (req, res) => {
   }
 
   rp(options)
-    .then((body) => {
-      body = JSON.parse(body);
+  .then((body) => {
+    body = JSON.parse(body);
+    if (body.ok) {
       let slackTeamData = {
         slackTeamToken: body.access_token,
         slackTeamName: body.team_name,
         slackTeamId: body.team_id,
         slackBotId: body.bot.bot_user_id,
         slackBotToken: body.bot.bot_access_token
       }
-
-      if (body.ok) {
-        Team.findOrCreate({ where : slackTeamData })
-          .spread( (team, created) => {
-            findTeamUsers(team);
-            res.redirect('/');
-          })
-          .catch(err => {
-            //TODO: redirect to error page due to bad data base request
-            res.redirect('/');
-          })
-      } else {
-        //TODO: redirect to error page due to failed respone request to slack
-        res.redirect('/');
-      }
-
-    })  
-    .catch(err => res.redirect('/') /* due to failed request to slack */ );
+      return slackTeamData;
+    } else {
+      //TODO: redirect to error page due to failed respone request to slack
+      res.redirect('/');
+    }
+  })
+  .then((slackTeamData) => {
+    Team.create(slackTeamData)
+    .then((team) => {
```

Comment:
nested promise is an anti-pattern. Return the team.create promise on line 35 and parallelize the .then chains